usrp_drivers/n200/n200_driver.cpp

```diff
@@ -382,245 +398,253 @@ std::string random_string( size_t length )
+  ERR_CHK_ZMQ(data_socket.connect("inproc://data"))
 
-  auto channels_set = false;
   auto center_freq_set = false;
 
-  uhd::time_spec_t time_zero;
-  uhd::rx_streamer::sptr rx_stream;
   uhd::stream_args_t stream_args("fc32", "sc16");
   uhd::stream_cmd_t stream_cmd(
       uhd::stream_cmd_t::STREAM_MODE_NUM_SAMPS_AND_DONE);
+  auto rx_rate_hz = driver_options.get_rx_rate();
 
-  auto rx_rate = driver_options->get_rx_rate();
-
-  auto receive_channels = usrp_d->get_receive_channels();
+  auto receive_channels = usrp_d.get_receive_channels();
+  stream_args.channels = receive_channels;
+  usrp_d.set_rx_rate(rx_rate_hz);  // ~450us
+  uhd::rx_streamer::sptr rx_stream = usrp_d.get_usrp_rx_stream(stream_args);  // ~44ms
 
-  computationpacket::ComputationPacket cp;
+  //This loop receives 1 pulse sequence worth of samples.
   while (1) {
-    packet_socket.recv(&request);
-    std::cout << "Received in RECEIVE\n";
-    driverpacket::DriverPacket dp;
-    std::string msg_str(static_cast<char*>(request.data()), request.size());
-    dp.ParseFromString(msg_str);
-
-    std::cout << "RECEIVE burst flags SOB " << dp.sob() << " EOB " << dp.eob() << std::endl;
-
-    if ( dp.sob() == false ) continue;
+    driver_packet_pub_socket.recv(&request);
+    DEBUG_MSG( "Received in RECEIVE");
+    driverpacket::DriverPacket driver_packet;
+    std::string packet_msg_str(static_cast<char*>(request.data()), request.size());
+    driver_packet.ParseFromString(packet_msg_str);
 
-    std::chrono::steady_clock::time_point stream_begin = std::chrono::steady_clock::now();
-    if (dp.channels_size() > 0 && dp.sob() == true && channels_set == false) {
-      stream_args.channels = receive_channels;
-      usrp_d->set_rx_rate(rx_rate);  // ~450us
-      rx_stream = usrp_d->get_usrp()->get_rx_stream(stream_args);  // ~44ms
-      channels_set = true;
-    }
-    std::chrono::steady_clock::time_point stream_end = std::chrono::steady_clock::now();
-    std::cout << "RECEIVE stream set up time: "
-      << std::chrono::duration_cast<std::chrono::microseconds>
-                                                  (stream_end - stream_begin).count()
-      << "us" << std::endl;
+    DEBUG_MSG("RECEIVE burst flags SOB " << driver_packet.sob() << " EOB " << driver_packet.eob());
 
+    if ( driver_packet.sob() == false ) continue;
 
-    std::chrono::steady_clock::time_point ctr_begin = std::chrono::steady_clock::now();
+    auto set_ctr_freq_begin = std::chrono::steady_clock::now();
 
-    if (dp.rxcenterfreq() > 0) {
-      std::cout << "RECEIVE center freq " << dp.rxcenterfreq() << std::endl;
-      usrp_d->set_rx_center_freq(dp.rxcenterfreq(), receive_channels);
+    if (driver_packet.rxcenterfreq() > 0) {
+      DEBUG_MSG("RECEIVE center freq " << driver_packet.rxcenterfreq()); // REVIEW #34 move this print statement below and use actual set rxcenterfreq
+      usrp_d.set_rx_center_freq(driver_packet.rxcenterfreq(), receive_channels);        // REPLY will discuss this.
```

Comment:
changed this up a bit. I changed the set_rx_center_freq function to return the actual frequency, but since that function loops over all channels Im making the assumption that all channels are updated the same. There is still a check to do something if the set freq is different from the intended one.